exchanges/bybit/bybit_websocket.go

```diff
@@ -0,0 +1,495 @@
+					case by.Websocket.DataHandler <- err:
+					default:
+						log.Errorf(log.WebsocketMgr,
+							"%s websocket handle data error: %v",
+							by.Name,
+							err)
+					}
+				}
+			default:
+			}
+			return
+		case resp := <-comms:
+			err := by.wsHandleData(resp.Raw)
+			if err != nil {
+				by.Websocket.DataHandler <- err
+			}
+		}
+	}
+}
+
+func (by *Bybit) wsHandleData(respRaw []byte) error {
+	var multiStreamData map[string]interface{}
+	err := json.Unmarshal(respRaw, &multiStreamData)
+	if err != nil {
+		return err
+	}
+
+	if method, ok := multiStreamData["event"].(string); ok {
+		if strings.EqualFold(method, sub) {
+			return nil
+		}
+		if strings.EqualFold(method, cancel) {
+			return nil
+		}
+	}
+
+	if e, ok := multiStreamData["e"].(string); ok {
+		switch e {
+		case wsAccountInfoStr:
+			var data wsAccountInfo
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to outboundAccountInfo structure %s",
+					by.Name,
+					err)
+			}
+			by.Websocket.DataHandler <- data
+			return nil
+		case wsOrderStr:
+			var data wsOrderUpdate
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to executionReport structure %s",
+					by.Name,
+					err)
+			}
+			var orderID = strconv.FormatInt(data.OrderID, 10)
+			oType, err := order.StringToOrderType(data.OrderType)
+			if err != nil {
+				by.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: by.Name,
+					OrderID:  orderID,
+					Err:      err,
+				}
+			}
+			var oSide order.Side
+			oSide, err = order.StringToOrderSide(data.Side)
+			if err != nil {
+				by.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: by.Name,
+					OrderID:  orderID,
+					Err:      err,
+				}
+			}
+			var oStatus order.Status
+			oStatus, err = stringToOrderStatus(data.OrderStatus)
+			if err != nil {
+				by.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: by.Name,
+					OrderID:  orderID,
+					Err:      err,
+				}
+			}
+
+			p, err := by.extractCurrencyPair(data.Symbol, asset.Spot)
+			if err != nil {
+				return err
+			}
+
+			by.Websocket.DataHandler <- &order.Detail{
+				Price:           data.Price,
+				Amount:          data.Quantity,
+				ExecutedAmount:  data.CumulativeFilledQuantity,
+				RemainingAmount: data.Quantity - data.CumulativeFilledQuantity,
+				Exchange:        by.Name,
+				ID:              orderID,
+				Type:            oType,
+				Side:            oSide,
+				Status:          oStatus,
+				AssetType:       asset.Spot,
+				Date:            data.OrderCreationTime.Time(),
+				Pair:            p,
+				ClientOrderID:   data.ClientOrderID,
+			}
+			return nil
+		case wsOrderFilledStr:
+			// already handled in wsOrderStr case
+			return nil
+		}
+	}
+
+	t, ok := multiStreamData["topic"].(string)
+	if !ok {
+		log.Errorf(log.ExchangeSys, "%s Received unhandle message on websocket: %s\n", by.Name, multiStreamData)
```

Comment:
There are a few auth and ping messages not being handled:

```json
[ERROR]|EXCHANGE| 27/04/2022 15:27:56 |Bybit Received unhandle message on websocket: map[ping:%!s(float64=1.651037280001e+12)]
```

please change %s -> %v

Reply:
I am unable to reproduce this. I am running the binary with config where WS is enabled. @shazbert 
Is there something else that you are doing here?

Reply:
I am running it with Ws enabled, Authentication enabled. 

Raw: 
```json 
{"auth":"success","userId":69696969}
```
Error here: 
```cmd
[ERROR]|EXCHANGE| 02/05/2022 12:48:58 |Bybit Received unhandle message on websocket: map[auth:success userId:69696969]
```

and for ping raw:
```json 
{"ping":1651459740001}
```

Reply:
I'm noting that ping still needs to be addressed:
`[ERROR] | WEBSOCKET |  05/05/2022 10:15:30  | exchange Bybit websocket error - unhandled stream data {"ping":1651709730003}`

Reply:
Thanks! Its working now